### PR TITLE
Check unique plan node ids

### PR DESCRIPTION
**Pull request by mbasmanova**
Plan node IDs must be unique. Add a check to detect duplicates and fail early at
task creation time to avoid difficult to troubleshoot runtime errors.

Enhance PlanBuilder to make it easy to ensure plan node IDs are unique. 

Introduce PlanNodeIdGenerator class that generates unique sequential node IDs
and modify PlanBuilder's constructor to take an instance of such generator.
When making non-linear plans using multiple instances of PlanBuilder just pass
the same instance of PlanNodeIdGenerator to each PlanBuilder to ensure all
nodes will have unique IDs.

### CODE REVIEW

File: velox/expression/VectorUdfTypeSystem.h [hunk 1]
```diff
@@ -1104,6 +1105,10 @@ struct VectorReader<Generic> {
     return GenericView{base_, index};
   }
 
+  exec_null_free_in_t readNullFree(vector_size_t offset) const {
```
Comment (aditi-pandit) [inline velox/expression/VectorUdfTypeSystem.h:1108]: Is this changed related to this PR ? On a quick look, doesn't seem like.

File: velox/exec/tests/HashJoinTest.cpp [hunk 2]
```diff
@@ -60,16 +60,18 @@ class HashJoinTest : public HiveConnectorTestBase {
...
-        PlanBuilder()
-            .values({leftBatch}, true)
-            .hashJoin(
-                makeKeyNames(keyTypes.size(), "t_"),
-                makeKeyNames(keyTypes.size(), "u_"),
-                PlanBuilder().values({rightBatch}, true).planNode(),
-                filter,
-                concat(leftType->names(), rightType->names()))
-            .planNode();
+    auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
```
Comment (aditi-pandit) [inline velox/exec/tests/HashJoinTest.cpp:63]: Move the planNodeIdGenerator to the class level so that it needn't be created at the tests level unless needed ? Same for other tests below.

**Review by aditi-pandit — CHANGES_REQUESTED** [review]

File: velox/expression/VectorUdfTypeSystem.h (hunk 1 above)
Comment (mbasmanova) [reply velox/expression/VectorUdfTypeSystem.h:1108]: @aditi-pandit This is an artifact of me stacking diffs in Phabricator, then exporting the top diff to GitHub. The PR ends up containing changes from 2 diffs in 2 commits.

File: velox/exec/tests/HashJoinTest.cpp (hunk 2 above)
Comment (mbasmanova) [reply velox/exec/tests/HashJoinTest.cpp:63]: It is not clear to me why this would be better. Currently, tests that need a generator make one and use it.
